miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/MiniAppDownloader.kt

```diff
@@ -123,6 +125,33 @@ internal class MiniAppDownloader(
         versionId: String
     ) = apiClient.fetchFileList(appId, versionId)
 
+    @Throws(MiniAppSdkException::class)
+    suspend fun fetchMiniAppManifest(appId: String, versionId: String): MiniAppManifest {
+        if (versionId.isEmpty()) throw MiniAppSdkException("Provided Mini App Version ID is invalid.")
+        else {
+            val manifestResponse = apiClient.fetchMiniAppManifest(appId, versionId)
+            return prepareMiniAppManifest(manifestResponse)
+        }
+    }
+
+    @VisibleForTesting
+    fun prepareMiniAppManifest(metadataEntity: MetadataEntity): MiniAppManifest {
+        try {
+            val requiredPermissions = metadataEntity.metadata?.requiredPermissions?.map {
+                Pair(MiniAppCustomPermissionType.getValue(it.name), it.reason)
+            } ?: emptyList()
+
+            val optionalPermissions = metadataEntity.metadata?.optionalPermissions?.map {
+                Pair(MiniAppCustomPermissionType.getValue(it.name), it.reason)
+            } ?: emptyList()
+
+            val customMetadata = metadataEntity.metadata?.customMetaData!!
```

Comment:
I don't think we should use the not-null assertion here. The mini app might not have this key, so we shouldn't throw an error in that case. Also, `manifest.json` is technically optional, so it's possible for the mini app to have an empty manifest.

Maybe this instead:
```suggestion
            val customMetadata = metadataEntity.metadata?.customMetaData ?: emptyMap()
```